enterprise/api/v1/views/analytics_summary.py

```diff
@@ -0,0 +1,58 @@
+"""
+Views for the `ChatGPTResponse` API endpoint.
+"""
+from edx_rbac.decorators import permission_required
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import generics, status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from django.shortcuts import get_object_or_404
+
+from enterprise.api.utils import (
+    generate_prompt_for_learner_engagement_summary,
+    generate_prompt_for_learner_progress_summary,
+)
+from enterprise.api.v1.serializers import AnalyticsSummarySerializer
+from enterprise.models import ChatGPTResponse, EnterpriseCustomer
+
+
+class AnalyticsSummaryView(generics.GenericAPIView):
+    """
+    API to generate a signed token for an enterprise admin to use Plotly analytics.
+    """
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = (IsAuthenticated,)
+
+    http_method_names = ['post']
+
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, enterprise_uuid: enterprise_uuid
+    )
+    def post(self, request, enterprise_uuid):
+        """
+        Generate auth token for plotly.
+        """
+        role = 'system'
+        enterprise_customer = get_object_or_404(EnterpriseCustomer, uuid=enterprise_uuid)
+
+        # Validate payload data
+        serializer = AnalyticsSummarySerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(
+                data={
+                    'errors': serializer.errors,
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
Instead of `if`, What about just
```python
serializer.is_valid(raise_exception=True)
```

Reply:
@muhammad-ammar If I use `raise_exception=True` then returned response does not contain `errors` key at the top level and response is

```
{
    'learner_progress': ['<error-message>'],
    'learner_engagement': ['<error-message>'],
}
```
this structure is same as the actual response

```
{
    'learner_progress': '<summary>',
    'learner_engagement':  '<summary>',
}
```

This could cause confusion, that is why I am using the `if` condition to change the structure of the response.